### ISSUE TITLE
Updating Params should occur one at a time

### ISSUE THREAD

**Issue by runyontr** [kudobuilder/terraform-provider-kudo#13]
Since KUDO doesn't currently support updating multiple parameters at the same time, we can use Terraform to control the rollout and update of the instance.

**Comment by alenkacz** [comment]
it does not? I think it does support that, only all changes have to go to the same trigger

**Comment by runyontr** [comment]
I believe you can update multiple params with different triggers, but only one of the plans runs.  With the provider we could do them one at a time since we have an understanding of the before/after state and can see which set of params are changing as the plans run, and which are still outstanding.

Do you think we should upgrade all params of the same trigger at the same time, or do them one at a time regardless of whether they're the same plan or not?

Order is something we might also need to consider

**Comment by alenkacz** [comment]
Yeah that is the behavior without webhook, with webhook, that request will be rejected.

Yeah that sounds like a good feature I just wonder how to determine the order in case of different triggers 🤔

**Comment by runyontr** [comment]
https://github.com/kudobuilder/kudo/blob/dc7492401f96f8a3a774bd2d07d867feb72b5671/pkg/webhook/instance_admission.go#L290-L321

Thank you for the lead.  I haven't had time to test with webhooks yet, but that's a great feature.  I think I would still want this feature so that in a GitOps work flow we don't have to have multiple commits, each with the partial change.

**Comment by alenkacz** [comment]
@runyontr yeah, that makes a ton of sense for gitops!
